pkg/virt-controller/watch/application.go

```diff
@@ -71,7 +83,21 @@ func Execute() {
 
 	app.restClient = app.clientSet.RestClient()
 
-	restful.Add(rest.WebService)
+	readinessFunc := func(_ *restful.Request, response *restful.Response) {
+		res := map[string]interface{}{}
+		if !app.isLeader {
```

Comment:
You have to also use locks here. The problem is, that ony the use of the same lock in both go-routines will force a syncrhonziation of shared variables between the two go-routines. Otherwise it can still happen that this go-routine will never see any update to app.isLeader. I think https://golang.org/ref/mem is a pretty good read to better understand that.

Reply:
:+1: and thanks for article